test/unit/3dtileslayerstyle.js

```diff
@@ -75,26 +76,26 @@ describe('3DTilesLayer Style', () => {
             view,
         );
 
-        $3dTilesLayer.style = {
-            fill: {
-                color: (c3DTileFeature) => {
-                    if (c3DTileFeature.batchId > 1) {
-                        return 'red';
-                    } else {
-                        return 'blue';
-                    }
-                },
-                opacity: (c3DTileFeature) => {
-                    if (c3DTileFeature.getInfo().something) {
-                        return 0.1;
-                    } else if (c3DTileFeature.userData.something === 'random') {
-                        return 1;
-                    } else {
-                        return 0.5;
-                    }
-                },
-            },
-        };
+        $3dTilesLayer.style = new Style(
+            {
+                fill: {
+                    color: (styleContextProperties, styleContext) => {
+                        if (styleContext.feature.batchId > 1) {
+                            return 'red';
+                        } else {
+                            return 'blue';
+                        }
+                    },
+                    opacity: (styleContextProperties, styleContext) => {
+                        if (styleContext.feature.getInfo().something) {
+                            return 0.1;
+                        } else if (styleContext.feature.userData.something === 'random') {
+                            return 1;
+                        } else {
+                            return 0.5;
+                        }
+                    } },
+            });
```

Comment:
I would prefer to keep the declarative way (i.e. an object) to declare a style, instead of an instance of `Style`. We have plans to keep the style class as internal to iTowns and not expose it anymore. Instead the style parameter shall respect the `Style` interface (or at least a subset of it in the case of `C3DTilesLayer`). You could initialize the `Style` class at `C3DTilesLayer` initialization.

Reply:
Ok, I've done a setter checking the type of style since ````C3DTilesLayer.style = value```` could be called in an itowns context ( ie  value is a Style instance) or by an itowns user (ie value is a literal object) if that's not the case and I'll remove the check and just treat the literal object. I also remove Style import from the widget 3dstyle since I've consider that widget code should be seen as itowns user code.